### PR TITLE
Cleanup runoff.c

### PR DESCRIPTION
**Pull request by tbohn**
This addresses issue #345.

This contains code changes from PR #347 (Feature/fix pet).

### CODE REVIEW

**Comment by jhamman** [comment]
@tbohn - can you rebase against develop.  You don't need the PET changes in here.

File: vic/vic_run/src/runoff.c [hunk 1]
```diff
@@ -48,8 +48,7 @@ runoff(cell_data_struct  *cell,
     size_t                     time_step;
     int                        last_index;
     int                        tmplayer;
-    size_t                     frost_area;
-    int                        fidx;
+    size_t                     fidx;
```
Comment (jhamman) [inline vic/vic_run/src/runoff.c]: See this related warning on the current build: https://travis-ci.org/UW-Hydro/VIC/jobs/103167674#L170-L174

Comment (tbohn) [inline vic/vic_run/src/runoff.c]: OK, I just now changed fidx from size_t to int.  That should fix the warning.

**Comment by tbohn** [comment]
I attempted to rebase relative to feature/fix_pet but the pull request is relative to the develop branch so the code differences from feature/fix_pet still show up.  I think once the PET PR is merged this one will be very straightforward.  It's not urgent - this can wait for the other one.

**Comment by jhamman** [comment]
No, you needed to `rebase` relative to `develop`, you are already "based" on `feature/fix_pet`.  Since these changes are not at all related, and do not build on each other at all, there is no reason to include them in the same PR.

**Pull request closed without merging**